cli/permissions.rs

```diff
@@ -354,3 +421,137 @@ fn permission_prompt(message: &str) -> DenoResult<PromptResult> {
     };
   }
 }
+
+fn check_path_white_list(
+  filename: &str,
+  white_list: &Arc<HashSet<String>>,
+) -> bool {
+  let mut path_buf = PathBuf::from(filename);
+
+  if white_list.contains(path_buf.to_str().unwrap()) {
+    return true;
+  }
+
+  while path_buf.pop() {
+    if white_list.contains(path_buf.to_str().unwrap()) {
+      return true;
+    }
+  }
+  false
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  #[test]
+  fn check_paths() {
+    let whitelist = vec![
+      "/a/specific/dir/name".to_string(),
+      "/a/specific".to_string(),
+      "/b/c".to_string(),
+    ].to_vec();
+
+    let perms = DenoPermissions::from_flags(&DenoFlags {
+      read_whitelist: whitelist.clone(),
+      write_whitelist: whitelist.clone(),
+      no_prompts: true,
+      ..Default::default()
+    });
+
+    // Inside of /a/specific and /a/specific/dir/name
+    perms.check_read("/a/specific/dir/name").unwrap();
```

Comment:
Rather than doing .unwrap() (which works) I think it would be more clear if you did

```rust
assert!(perms.check_read("/a/specific/dir/name").is_ok());
```